test/adaptive_load/scoring_function_test.cc

```diff
@@ -0,0 +1,231 @@
+}
+
+TEST(BinaryScoringFunctionTest, ReturnsNegativeOneForValueBelowThresholdRange) {
+  nighthawk::adaptive_load::BinaryScoringFunctionConfig config;
+  config.mutable_lower_threshold()->set_value(5.0);
+  config.mutable_upper_threshold()->set_value(7.0);
+  BinaryScoringFunction scoring_function(config);
+  EXPECT_EQ(scoring_function.EvaluateMetric(4.0), -1.0);
+}
+
+TEST(BinaryScoringFunctionTest, ReturnsNegativeOneForValueAboveThresholdRange) {
+  nighthawk::adaptive_load::BinaryScoringFunctionConfig config;
+  config.mutable_lower_threshold()->set_value(5.0);
+  config.mutable_upper_threshold()->set_value(7.0);
+  BinaryScoringFunction scoring_function(config);
+  EXPECT_EQ(scoring_function.EvaluateMetric(8.0), -1.0);
+}
+
+TEST(LinearScoringFunctionTest, ReturnsZeroForValueEqualToThreshold) {
+  nighthawk::adaptive_load::LinearScoringFunctionConfig config;
+  config.set_threshold(10.0);
+  config.set_k(1.0);
+  LinearScoringFunction scoring_function(config);
+  EXPECT_EQ(scoring_function.EvaluateMetric(10.0), 0.0);
+}
+
+TEST(LinearScoringFunctionTest, ReturnsPositiveValueForValueBelowThreshold) {
+  nighthawk::adaptive_load::LinearScoringFunctionConfig config;
+  config.set_threshold(10.0);
+  config.set_k(1.0);
+  LinearScoringFunction scoring_function(config);
+  EXPECT_EQ(scoring_function.EvaluateMetric(8.0), 2.0);
+}
+
+TEST(LinearScoringFunctionTest, ReturnsNegativeValueForValueAboveThreshold) {
+  nighthawk::adaptive_load::LinearScoringFunctionConfig config;
+  config.set_threshold(10.0);
+  config.set_k(1.0);
+  LinearScoringFunction scoring_function(config);
+  EXPECT_EQ(scoring_function.EvaluateMetric(12.0), -2.0);
+}
+
+TEST(SigmoidScoringFunctionTest, ReturnsZeroForValueEqualToThreshold) {
+  nighthawk::adaptive_load::SigmoidScoringFunctionConfig config;
+  config.set_threshold(10.0);
+  config.set_k(1.0);
+  SigmoidScoringFunction scoring_function(config);
+  EXPECT_EQ(scoring_function.EvaluateMetric(10.0), 0.0);
+}
+
+TEST(SigmoidScoringFunctionTest, ReturnsPositiveValueForValueBelowThreshold) {
+  nighthawk::adaptive_load::SigmoidScoringFunctionConfig config;
+  config.set_threshold(1000.0);
+  config.set_k(0.001);
+  SigmoidScoringFunction scoring_function(config);
+  EXPECT_GT(scoring_function.EvaluateMetric(999.0), 0.0);
+}
+
+TEST(SigmoidScoringFunctionTest, ReturnsNegativeValueForValueAboveThreshold) {
+  nighthawk::adaptive_load::SigmoidScoringFunctionConfig config;
+  config.set_threshold(1000.0);
+  config.set_k(0.001);
+  SigmoidScoringFunction scoring_function(config);
+  EXPECT_LT(scoring_function.EvaluateMetric(1001.0), 0.0);
+}
+
+TEST(SigmoidScoringFunctionTest, ReturnsValueCloseToOneForValueFarBelowThreshold) {
+  nighthawk::adaptive_load::SigmoidScoringFunctionConfig config;
+  config.set_threshold(1000.0);
+  config.set_k(0.001);
+  SigmoidScoringFunction scoring_function(config);
+  const double score = scoring_function.EvaluateMetric(-10000.0);
+  EXPECT_GT(score, 0.99);
```

Comment:
There's also EXPECT_NEAR. Not sure if it's needed here, but in the statistics test we also have a helper which facilitates verifying expected precision against output of a reference implementation:

https://github.com/envoyproxy/nighthawk/blob/master/test/statistic_test.cc#L180

Reply:
Deleted these tests, will remember EXPECT_NEAR.